### DISCUSSION TITLE
RateLimitError when using text-embedding-ada-002 from OpenAI

### DISCUSSION THREAD

**Discussion by chonchonj23** [MaartenGr/BERTopic#1434] [Q&A]
Dear Maarten,

I tried to use OpenAI's text-embedding-ada-002, but got below RateLimitError. Any trick I can get around this?

![RateLimitError](https://github.com/MaartenGr/BERTopic/assets/22134203/9dfdbff3-9675-4f60-a7e9-3c611c17e632)

Thank you very much for your help.

Best regards,
Chon

**Comment by MaartenGr** [comment]
That is most likely since you have reached your token limit with your account. If you exceed it, then you cannot continue accessing the API until you fill up again or pay more. You could also use the `delay_in_seconds` parameter to make sure that you do not request the API too much within each minute, thereby remaining below your assigned rate limit.

**Reply by chonchonj23** [reply]
Thanks Maarten!

My code works after adding the delay_in_seconds and batch_size parameter:
embedding_model = OpenAIBackend("text-embedding-ada-002", delay_in_seconds=5, batch_size=800)
